driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -221,7 +231,9 @@ public Connection borrowConnection(long timeout, TimeUnit unit) throws Connectio
         }
 
         int connectionCount = open.get() + scheduledForCreation.get();
-        if (connectionCount < options().getMaxConnectionsPerHost(hostDistance)) {
+        if (connectionCount < options().getCoreConnectionsPerHost(hostDistance)) {
+            maybeSpawnNewConnection();
```

Comment:
This would only spawn 1 new connection, not up to core connections.  If I have a scenario where I have a Core Pool Size for 8 and 4 of the 8 connections are established successfully on pool init, this will spawn 1 new connection instead of 4.  However, if I were to make 4 requests it would get me up to 8 connections, so to me that doesn't seem like we need any special logic for this, since if we don't end up making up to core pool size requests on a Host, we don't need up to core pool size connections!